core/src/main/java/io/github/fourlastor/game/demo/round/step/ChargeBeam.kt

```diff
@@ -0,0 +1,47 @@
+package io.github.fourlastor.game.demo.round.step
+
+import com.badlogic.gdx.scenes.scene2d.actions.Actions
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedFactory
+import dagger.assisted.AssistedInject
+import io.github.fourlastor.game.demo.state.GameState
+import io.github.fourlastor.game.demo.state.unit.Mon
+import io.github.fourlastor.game.demo.state.unit.effect.ChargeBeamEffect
+
+class ChargeBeam @AssistedInject constructor(
+    @Assisted("source") private val source: Mon,
+    chargeBeamFactory_: ChargeBeamEffect.Factory
```

Comment:
you can have `private val chargeBeamFactory: ChargeBeamEffect.Factory` and remove the `init` block and the variable declared in the body